src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/SecurityExtensionTest.php

```diff
@@ -166,8 +166,7 @@ public function testPerListenerProvider()
     }
 
     /**
-     * @expectedException \Symfony\Component\Config\Definition\Exception\InvalidConfigurationException
-     * @expectedExceptionMessage Not configuring explicitly the provider for the "http_basic" listener on "ambiguous" firewall is ambiguous as there is more than one registered provider.
+     * @expectedException \TypeError
```

Comment:
getting a `TypeError` when not setting the provider isn't fine to me as this aims to make it optional

Reply:
Yes I agree we need a proper error but I don't know when to throw it.